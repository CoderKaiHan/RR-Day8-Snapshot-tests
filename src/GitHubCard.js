import React from 'react';
import { LoremIpsum } from 'react-lorem-ipsum';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';

function GitHubCard () {
    return (
        <div>
          <Card style={{ width: '400px', margin: 'auto' }}>
            <Card.Img variant="top" src="https://avatars.githubusercontent.com/u/147564905?s=400&u=4bd21078a4e3e930415da3b001dd4c1127e03e90&v=4" />
            <Card.Body>
              <Card.Title style={{fontWeight:'bold'}}>GitHub name: CoderKaiHan</Card.Title>
              <Card.Text>
              <LoremIpsum p={1} />
              </Card.Text>
              <Card.Link href='https://github.com/CoderKaiHan' target="_blank" rel="noopener noreferrer" ><Button variant="primary">Go to GitHub</Button></Card.Link>
            </Card.Body>
          </Card>
        </div>
    );

}

export default GitHubCard;