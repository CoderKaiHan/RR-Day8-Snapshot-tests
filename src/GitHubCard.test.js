import renderer from 'react-test-renderer';
import GitHubCard from './GitHubCard';


test('Renders GithubCard snapshot', () => {
    const tree = renderer.create(<GitHubCard />).toJSON()
    expect(tree).toMatchSnapshot()
  })
