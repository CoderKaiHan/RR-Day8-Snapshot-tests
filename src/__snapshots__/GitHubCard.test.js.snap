// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders GithubCard snapshot 1`] = `
<div>
  <div
    className="card"
    style={
      Object {
        "margin": "auto",
        "width": "400px",
      }
    }
  >
    <img
      className="card-img-top"
      src="https://avatars.githubusercontent.com/u/147564905?s=400&u=4bd21078a4e3e930415da3b001dd4c1127e03e90&v=4"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        GitHub name: CoderKaiHan
      </div>
      <p
        className="card-text"
      >
        <p>
          Lorem ipsum odor amet, consectetuer adipiscing elit. Bibendum risus congue scelerisque, nullam odio quis rhoncus libero molestie. Adapibus mi convallis dictum leo; pulvinar tortor. Sit felis nisi litora tellus, lacus nulla imperdiet? Dui fusce imperdiet vitae enim a curae ornare. Ad sagittis blandit mattis pulvinar sed. Nibh pharetra volutpat quis ultrices metus volutpat lectus accumsan.
        </p>
      </p>
      <a
        className="card-link"
        href="https://github.com/CoderKaiHan"
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          Go to GitHub
        </button>
      </a>
    </div>
  </div>
</div>
`;
